### PR TITLE
Maternodes Upgrade

### PR DESCRIPTION
**Pull request by BlockMechanic**
This is a set of commits that implements a Network upgrade that includes Masternodes

### CODE REVIEW

File: src/consensus/upgrades.cpp [hunk 1]
```diff
@@ -30,8 +30,8 @@ const struct NUInfo NetworkUpgradeInfo[Consensus::MAX_NETWORK_UPGRADES] = {
         /*.strInfo =*/ "See https://z.cash/upgrade/sapling.html for details.",
     },
     {
-        /*.nBranchId =*/ 0x76b809bb,
-        /*.strName =*/ "Alfheimr",
+        /*.nBranchId =*/ 0x736c627a,
+        /*.strName =*/ "Bzedge",
```
Comment (faneaatiku) [inline src/consensus/upgrades.cpp]: This should be BZShares

File: src/chainparams.cpp [hunk 2]
```diff
@@ -261,11 +267,14 @@ class CMainParams : public CChainParams {
...
-        vFoundersRewardAddress = {};
-        assert(vFoundersRewardAddress.size() <= consensus.GetLastFoundersRewardBlockHeight());
+        nPoolMaxTransactions = 3;
+        strSporkKey = "045da9271f5d9df405d9e83c7c7e62e9c831cc85c51ffaa6b515c4f9c845dec4bf256460003f26ba9d394a17cb57e6759fe231eca75b801c20bccd19cbe4b7942d";
+
+        strObfuscationPoolDummyAddress = "t1cW3eB2pruAMdfc7nu5nSbEcRqdGNMup3s";
+        nStartMasternodePayments = 1574683200; //2018-04-15
+        nBudget_Fee_Confirmations = 6; // Number of confirmations for the finalization fee
+        masternodeProtectionBlock = 883000;
+        masternodeCollateral = 10000;
```
Comment (faneaatiku) [inline src/chainparams.cpp]: This should be 250000

File: src/consensus/upgrades.cpp (hunk 1 above)
Comment (BlockMechanic) [reply src/consensus/upgrades.cpp]: Done, must have been reverted in an earlier change.

File: src/chainparams.cpp (hunk 2 above)
Comment (BlockMechanic) [reply src/chainparams.cpp]: Ths too

**Pull request merged**